src/execution/operator/persistent/base_csv_reader.cpp

```diff
@@ -88,6 +108,36 @@ bool BaseCSVReader::TryCastValue(const Value &value, const LogicalType &sql_type
 		string error_message;
 		return options.date_format[LogicalTypeId::TIMESTAMP].TryParseTimestamp(string_t(StringValue::Get(value)),
 		                                                                       result, error_message);
+	} else if (options.decimal_separator != "." && sql_type.id() == LogicalTypeId::DECIMAL) {
+		string_t value_str = string_t(StringValue::Get(value));
+		auto width = DecimalType::GetWidth(sql_type);
+		auto scale = DecimalType::GetScale(sql_type);
+		switch (sql_type.InternalType()) {
+		case PhysicalType::INT8:
+			return TryCastDecimalOperator::Operation<TryCastToDecimalCommaSeparated, int8_t>(value_str, width, scale);
```

Comment:
Decimals cannot be stored in `INT8`, only in `16, 32, 64, 128` - so this switch case can be removed.